### ISSUE TITLE
XXE in XmlFile.java

### ISSUE THREAD

**Issue by ProDigySML** [OpenGamma/Strata#1796]
# The Issue

An XML External Entity attack is a type of attack against an application that parses XML input. This attack occurs when XML input containing a reference to an external entity is processed by a weakly configured XML parser. This attack may lead to the disclosure of confidential data, denial of service, server side request forgery, port scanning from the perspective of the machine where the parser is located, and other system impacts.

# Where the Issue Occurred

The following code snippets display the usage of `XMLInputFactory` without securely disabling entities:

https://github.com/OpenGamma/Strata/blob/eb76acd9c73c7968473574cd69fda13acffc0675/modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlFile.java#L92-L95

**Comment by jodastephen** [comment]
I don't think this is an issue because the code at line 92 calls `xmlInputFactory()`, and that method sets entity and DTD parsing off:
https://github.com/OpenGamma/Strata/blob/eb76acd9c73c7968473574cd69fda13acffc0675/modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlFile.java#L188-L200

Happy to receive feedback if `xmlInputFactory()` is incorrect.
